packages/contracts/token_contract/tests/utils/mod.rs

```diff
@@ -0,0 +1,186 @@
+    pub coins_per_wallet: Option<u64>,
+    pub coin_amount: Option<u64>,
+    pub mint_price: Option<u64>,
+    pub mint_limit: Option<u64>,
+    pub total_supply: Option<u64>,
+}
+
+pub struct ConstructorParams {
+    pub mint_price: u64,
+    pub mint_limit: u64,
+    pub total_supply: u64,
+}
+
+pub struct Metadata {
+    pub contract_id: Bech32ContractId,
+    pub asset_id: ContractId,
+    pub token_contract: TokenContract,
+    pub deployer: WalletUnlocked,
+    pub user: WalletUnlocked,
+    pub params: ConstructorParams,
+    pub provider: Provider,
+}
+
+pub mod abi_calls {
+    use super::*;
+    use fuels::programs::call_response::FuelCallResponse;
+
+    pub async fn constructor(
+        contract: &TokenContract,
+        params: &ConstructorParams,
+    ) -> FuelCallResponse<()> {
+        contract
+            .methods()
+            .constructor(params.mint_price, params.mint_limit, params.total_supply)
+            .call()
+            .await
+            .unwrap()
+    }
+
+    pub async fn owner(contract: &TokenContract) -> FuelCallResponse<Identity> {
+        contract.methods().owner().call().await.unwrap()
+    }
+    pub async fn mint_limit(contract: &TokenContract) -> FuelCallResponse<u64> {
+        contract.methods().mint_limit().call().await.unwrap()
+    }
+    pub async fn mint_price(contract: &TokenContract) -> FuelCallResponse<u64> {
+        contract.methods().mint_price().call().await.unwrap()
+    }
+    pub async fn total_minted(contract: &TokenContract) -> FuelCallResponse<u64> {
+        contract.methods().total_minted().call().await.unwrap()
+    }
+    pub async fn total_supply(contract: &TokenContract) -> FuelCallResponse<u64> {
+        contract.methods().total_supply().call().await.unwrap()
+    }
+    pub async fn coin_balance(
+        contract: &TokenContract,
+        asset_id: AssetId,
+    ) -> FuelCallResponse<u64> {
+        let contract_id = ContractId::new(asset_id.into());
+        contract
+            .methods()
+            .coin_balance(contract_id)
+            .call()
+            .await
+            .unwrap()
+    }
+    pub async fn deposit_of(contract: &TokenContract, id: Identity) -> FuelCallResponse<u64> {
+        contract.methods().deposit_of(id).call().await.unwrap()
+    }
+    pub async fn mint_of(contract: &TokenContract, id: Identity) -> FuelCallResponse<u64> {
+        contract.methods().mint_of(id).call().await.unwrap()
+    }
+
+    pub async fn deposit(
+        contract: &TokenContract,
+        wallet: WalletUnlocked,
+        amount: u64,
+    ) -> FuelCallResponse<u64> {
+        let tx_params = TxParameters::new(None, Some(1_000_000), None);
+        let call_params = CallParameters::new(Some(amount), Some(BASE_ASSET_ID), Some(1_000_000));
+
+        contract
+            .with_wallet(wallet)
+            .unwrap()
+            .methods()
+            .deposit()
+            .tx_params(tx_params)
+            .call_params(call_params)
+            .expect("REASON")
```

Comment:
I would change the message to be more specific so that you know where it came from e.g. `deposit failed`. This tells you the call that failed although `failed` is not exactly informative